iohub/convert.py

```diff
@@ -357,7 +357,7 @@ def _gen_chunks(self, input_chunks):
         # it's OK if a single image is larger than MAX_CHUNK_SIZE
         while (
             chunks[-3] > 1
-            and np.prod(chunks) * bytes_per_pixel > MAX_CHUNK_SIZE
+            and np.prod(chunks, dtype=float) * bytes_per_pixel > MAX_CHUNK_SIZE
```

Comment:
For example I would expect this to be sufficient for any array that can fit on our data partitions:

```suggestion
            and np.prod(chunks, dtype=np.int64) * bytes_per_pixel > MAX_CHUNK_SIZE
```